### PR TITLE
Tests: Crate/Primitives tests

### PR DESCRIPTION
**Pull request by danilowhk**
Added tests for:

- crate/primitives/block
- crate/primitives/crypto
- crate/primitives/execution
- crate/primitives/state
- crate/primitives/transaction

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing
- [ ] Other (please describe):

# What is the current behavior?

Issue Number: N/A

# What is the new behavior?

-
-
-

# Does this introduce a breaking change?

- [ ] Yes
- [ ] No

# Other information

### CODE REVIEW

File: crates/primitives/starknet/tests/block.rs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        let header = generate_dummy_header();
+        let expected_hash = H256::from_slice(
+            frame_support::Hashable::blake2_256(&header.block_number.encode()).as_slice(),
+        );
+
+        assert_eq!(header.hash(), expected_hash);
+    }
+
+    #[test]
+    fn test_header_clone() {
```
Comment (tdelabro) [inline crates/primitives/starknet/tests/block.rs:54]: Useless. You are testing the derive implem of clone. It's in the core of the Rust language. You will have to trust it.

File: crates/primitives/starknet/tests/block.rs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        let header = generate_dummy_header();
+        let cloned_header = header.clone();
+
+        assert_eq!(header, cloned_header);
+    }
+
+    #[test]
+    fn test_header_default() {
```
Comment (tdelabro) [inline crates/primitives/starknet/tests/block.rs:62]: Also testing the language itself. Not usefull

File: crates/primitives/starknet/tests/block.rs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            H256::from_slice(&[3; 32]),
+            0,
+            H256::from_slice(&[4; 32]),
+            Some(1),
+            Some(U256::from(3)),
+        )
+    }
+
+    #[test]
+    fn test_header_creation() {
```
Comment (tdelabro) [inline crates/primitives/starknet/tests/block.rs:27]: You are testing the `new` function there. Not really useful a long as it does not contain any custom logic.
If you want you can use `derive_more::Constructor` instead, so you just have to trust the lib and not test it

File: crates/primitives/starknet/tests/state.rs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    let nonce = Nonce::default();
+
+    state.address_to_nonce.insert(address, nonce);
+
+    let result = state.get_nonce_at(address).unwrap();
+    assert_eq!(result, nonce);
+}
+
+#[test]
+fn test_get_contract_class() {
```
Comment (tdelabro) [inline crates/primitives/starknet/tests/state.rs:37]: check the error case also: `None => Err(StateError::UndeclaredClassHash(*class_hash)),`

**Review by tdelabro — DISMISSED** [review]

**Comment by danilowhk** [comment]
> @danilowhk Do you need more help on this one? Or is it ready?

The reviews are fine :) Will update make some last changes and submit as PR, thank you @tdelabro !

File: crates/primitives/starknet/tests/block.rs [hunk 5]
```diff
@@ -0,0 +1,80 @@
+
+
+#[cfg(test)]
+mod tests {
```
[the remark this answers is not in the document]
Comment (danilowhk) [reply crates/primitives/starknet/tests/block.rs:2]: Do I add on src/lib.rs?

**Comment by danilowhk** [comment]
Changes to coverage are still to small will add more tests

File: crates/primitives/starknet/tests/block.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (abdelhamidbakhta) [reply crates/primitives/starknet/tests/block.rs:2]: @tdelabro

**Comment by abdelhamidbakhta** [comment]
hey @danilowhk what is the status of this PR ?

**Review by abdelhamidbakhta — APPROVED** [review]
lgtm
@danilowhk merging this one since we want to clear the backlog of pending PRs, but feel free to submit follow up PRs to add moarrrr tests

**Pull request merged**